benchmark/Streamly/Benchmark/Data/Array/Foreign/Mut.hs

```diff
@@ -136,21 +140,27 @@ composeN n f x =
         4 -> f x >>= f >>= f >>= f
         _ -> undefined
 
+scanls' :: Monad m => (b -> a -> b) -> b -> Stream.Stream m a -> Stream.Stream m b
+scanls' step z m = Stream.fromStreamD $ D.scanl' step z (Stream.toStreamD m)
+
+scanls1' :: Monad m => (a -> a -> a) -> Stream.Stream m a -> Stream.Stream m a
+scanls1' f m =  Stream.fromStreamD $ D.scanl1' f (Stream.toStreamD m)
```

Comment:
We cannot use StreamD functions here unless we want to specifically test StreamD functions for benchmark comparison or for some other reason. StreamD is an internal module, this is not how anyone will use the library.

We can continue using Streamly.Prelude functions or use folds to scan.